src/instrumentation/do-rpc.ts

```diff
@@ -0,0 +1,249 @@
+import { context as api_context, trace, SpanOptions, SpanKind, Exception, SpanStatusCode, Attributes, Context, propagation } from '@opentelemetry/api'
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions'
+import {  unwrap, wrap } from '../wrap.js'
+import { instrumentEnv } from './env.js'
+import { Initialiser, setConfig, getActiveConfig } from '../config.js'
+import { exportSpans } from './common.js'
+import { DOConstructorTrigger } from '../types.js'
+
+type AlarmFn = DurableObject['alarm']
+type Env = Record<string, unknown>
+type AnyFn = (...args: any[]) => any
+
+const dbSystem = 'Cloudflare DO SQLite'
+
+
+export function getParentContextFromHeaders(headers: Headers): Context {
+	return propagation.extract(api_context.active(), headers, {
+		get(headers, key) {
+			return headers.get(key) || undefined
+		},
+		keys(headers) {
+			return [...headers.keys()]
+		},
+	})
+}
+
+function getParentContextFromRequest(request: Request) {
+	const workerConfig = getActiveConfig()
+
+	if (workerConfig === undefined) {
+		return api_context.active()
+	}
+
+	const acceptTraceContext =
+		typeof workerConfig.handlers.fetch.acceptTraceContext === 'function'
+			? workerConfig.handlers.fetch.acceptTraceContext(request)
+			: (workerConfig.handlers.fetch.acceptTraceContext ?? true)
+	return acceptTraceContext ? getParentContextFromHeaders(request.headers) : api_context.active()
+}
+
+function spanOptions(dbName: string, operation: string, sql?: string): SpanOptions {
+	const attributes: Attributes = {
+		binding_type: 'DO',
+		[SemanticAttributes.DB_NAME]: dbName,
+		[SemanticAttributes.DB_SYSTEM]: dbSystem,
+		[SemanticAttributes.DB_OPERATION]: operation,
+	}
+	if (sql) {
+		attributes[SemanticAttributes.DB_STATEMENT] = sql?.replace(/\n/g, ' ').replace(/\s+/g, ' ').trim();
+	}
+	return {
+		kind: SpanKind.CLIENT,
+		attributes,
+	}
+}
+
+function metaAttributes(meta: SqlStorageIngestResult): Attributes {
+	return {
```

Comment:
Ideally, you can return the entire `meta` here. Just converting camelCase to snake case.

Reply:
The return value (result) is a Cursor. The keys aren't iterable.